element_calcium_imaging/imaging.py

```diff
@@ -165,13 +165,36 @@ def make(self, key):
                     'data_path': [tiff_files[0].parent.as_posix()],
                     'tiff_list': [f.as_posix() for f in tiff_files]
                 }
-                
+
                 suite2p.run_s2p(ops=suite2p_params, db=suite2p_paths)  # Run suite2p
 
                 _, imaging_dataset = get_loader_result(key, ProcessingTask)
                 suite2p_dataset = imaging_dataset
                 key = {**key, 'processing_time': suite2p_dataset.creation_time}
 
+            elif method == 'caiman':
+                from element_interface.run_caiman import run_caiman
+
+                tiff_files = (ProcessingTask * scan.Scan * scan.ScanInfo * scan.ScanInfo.ScanFile & key).fetch('file_path')
+                tiff_files = [find_full_path(get_imaging_root_data_dir(), tiff_file) for tiff_file in tiff_files]
+                tiff_files = [f.as_posix() for f in tiff_files]
+
+                params = (ProcessingTask * ProcessingParamSet & key).fetch1('params')
+                sampling_rate = (ProcessingTask * scan.Scan * scan.ScanInfo & key).fetch1('fps')
+                output_dir = (ProcessingTask & key).fetch1('processing_output_dir')
+                output_dir = find_full_path(get_imaging_root_data_dir(), output_dir).as_posix()
+
+                ndepths = (ProcessingTask * scan.Scan * scan.ScanInfo & key).fetch1('ndepths')
+
+                is3D = True if ndepths > 1 else False
+                if is3D:
+                    raise NotImplementedError('Caiman pipeline is not capable of analyzing 3D scans at the moment.')
```

Comment:
Is this error handing really necessary? Can you just catch the exception raised by `run_caiman`?

Reply:
These are two comments from the `caiman` pipeline:
> TODO - motion_correct_3d needs to be implemented in movies.py
> motion_correct_3d has not been implemented yet - instead initialize to just median image

Additionally, when I tried running the caiman pipeline on a 3D scan data, the code neither throw an exception nor was able to complete the run in an hour for a very small dataset (I had to cancel the run). Asa a result I came to the conclusion that we should add this error handling and take a deeper look into this later.